shelley/chain-and-ledger/formal-spec/leader-value.tex

```diff
@@ -0,0 +1,59 @@
+\section{Leader Value Calculation}
+\label{sec:leader-value-calc}
+
+This section details how we determine whether a node is entitled to lead (under
+the Praos protocol) given the output of its verifiable random function
+calculation.
+
+\begin{figure}
+  \emph{Values associated with the leader value calculations}
+  \begin{equation*}
+  \begin{array}{r@{~\in~}lr}
+    \var{certNat} & \{n | n \in \N, n \in [0,2^{256})\} & \text{Certified natural value from VRF} \\
+    \var{f} & [0,1] & \text{Active slot coefficient} \\
+    \sigma & [0,1] & \text{Stake proportion}
+  \end{array}
+  \end{equation*}
+\end{figure}
+
+\subsection{Computing the leader value}
+
+The verifiable random function gives us a 32-byte random output. We interpret
+this as a natural number $\var{certNat}$ in the range $[0,2^{256})$.
+
+\subsection{Node eligibility}
+
+As per \cite{ouroboros_praos}, a node is eligible to lead when its leader value
+$\ell < 1 - (1 - f)^\sigma$. We have
+
+\begin{align*}
+  \ell & < 1 - (1 -f)^\sigma \\
+  \iff \left(\frac{1}{1-\ell}\right) & > \exp{(-\sigma \cdot \ln{(1-f)})}
+\end{align*}
+
+The latter inquality can be efficiently computed through use of its Taylor
+expansion and error estimation to stop computing terms once we are certain that
+the result will be either above or below the target value.
+
+We carry out all computations using fixed precision arithmetic (specifically, we
+use 34 decimal bits of precision, since this is enough to represent the fraction
+of a single lovelace.)
+
+As such, we define the following:
+
+\begin{align*}
+  q & = \frac{2^{256}}{2^{256} - \var{certNat}} \\
+  c & = \ln{(1 - f)}
+\end{align*}
+
+and define the function \textit{checkLeaderVal} as follows:
+
+\begin{equation*}
+  \fun{checkLeaderVal}~\var{certNat}~\sigma~\var{f} =
+    \left\{
+      \begin{array}{l@{~}r}
+        \mathsf{True}, & f = 1 \\
```

Comment:
we don't have to do this in this PR, but I wonder if we should just ban `f=1`. or did we decide to keep it for testing purposes?

Reply:
If `f=1`, then every node leads every slot, which is anathema to the design of Praos. I at least don't have any plans for the `ThreadNet` tests to investigate `f=1`.

Maybe some lower-level tests could find it useful.